### PR TITLE
feat: Retry the friends initialization

### PR DESCRIPTION
**Pull request by sandrade-dcl**
# What? 
Now the initialization of the friends feature is being retried when it fails. That retry is carried out using an exponential time increment up to a maximum allowed.

### CODE REVIEW

**Comment by menduz** [comment]
To prevent unnecessary messages and moving the control to the renderer, I'd implement this as:

```typescript
while(true) {
try {
  yield call(initialize)
} catch {
  if (attempt++ < MAX_ATTEMPTS) {
    yield delay(N)
  } else {
    break
  }
}
```

File: packages/shared/friends/sagas.ts [hunk 1]
```diff
@@ -75,13 +78,25 @@ function* initializeFriendsSaga() {
...
-      const synapseUrl: string = yield select(getSynapseUrl)
-      yield call(initializePrivateMessaging, synapseUrl, identity)
-    } catch (e) {
-      logger.error(`error initializing private messaging`, e)
+    let secondsToRetry = MIN_TIME_BETWEEN_FRIENDS_INITIALIZATION_RETRIES_MILLIS
+
+    while(true) {
+      try {
+        const synapseUrl: string = yield select(getSynapseUrl)
+        yield call(initializePrivateMessaging, synapseUrl, identity)
```
Comment (menduz) [inline packages/shared/friends/sagas.ts:86]: when does this loop ends?

Comment (menduz) [reply packages/shared/friends/sagas.ts:86]: but it doesn't stop if it worked, or I am missing it

Comment (menduz) [reply packages/shared/friends/sagas.ts:86]: Here are two examples of controlled infinite loops https://github.com/decentraland/kernel/blob/chore/new-comms-sagas/packages/shared/comms/sagas.ts#L114-L154

Comment (menduz) [reply packages/shared/friends/sagas.ts:86]: I'd add something like 
```ts
const client: SocialAPI | null = yield select(getClient)
const isLoggedIn: boolean = client && (yield apply(client, client.isLoggedIn) || false
if (isLoggedIn) {
  // reset timers
  // yield delay(initialDelay)
} else {
  yield call(initializePrivateMessaging, synapseUrl, identity)
}
```

File: packages/shared/friends/sagas.ts [hunk 2]
```diff
@@ -48,6 +48,7 @@ import { ensureFriendProfile } from './ensureFriendProfile'
 import { getSynapseUrl } from 'shared/meta/selectors'
 import { store } from 'shared/store/isolatedStore'
 import { trackEvent } from '../analytics'
+import { apply } from 'redux-saga-test-plan/matchers'
```
Comment (menduz) [inline packages/shared/friends/sagas.ts:51]:
```suggestion
import { apply } from 'redux-saga/effects'
```

Comment (sandrade-dcl) [reply packages/shared/friends/sagas.ts]: Ups, fixed!

**Review by menduz — DISMISSED** [review]

File: packages/shared/friends/sagas.ts [hunk 3]
```diff
@@ -26,158 +24,166 @@ import {
...
+    yield call(waitForRealmInitialized)
+    yield call(waitForRendererInstance)
+
+    const currentIdentity: ExplorerIdentity | undefined = yield select(getCurrentIdentity)
+    const client: SocialAPI | null = yield select(getSocialClient)
+    const isLoggedIn: boolean = (currentIdentity && client && (yield apply(client, client.isLoggedIn, []))) || false
+
+    const shouldRetry = !isLoggedIn
+
+    if (shouldRetry) {
```
Comment (moliva) [inline packages/shared/friends/sagas.ts:106]: Should we have a limited number of retries instead of doing it indefinitely?

Comment (menduz) [reply packages/shared/friends/sagas.ts:106]: The idea was to have it indefinitely recovery. but the client has a problem that prevents it from happening if some events are pending to be sent to matrix. I'd recommend to stop using the dcl-social-api and using matrix-js-sdk from kernel directly instead

**Pull request merged**